### ISSUE TITLE
SingleFile, SingleFileZ, both Chrome and Firefox, suddenly failing on Twitter

### ISSUE THREAD

**Issue by JimOutsideDC** [gildas-lormeau/SingleFile#542]
Hello Gildas - We chatted back in August when I was having problems with Firefox.  I have a new problem.  As of yesterday, I can not get SingleFile or SingleFileZ to save any more than 2.5 MB of my Twitter feed, even displaying hours of tweets.  Two months ago, SingleFile saved one file that was 120 MB.  More recently, saving 10-15 hours of my Twitter timeline feed produced SingleFile saves typically 70-75 MB, so until now, it seemed to be working fine.  Suddenly, if it save 2.5 MB, that's it. Or else it hangs.  It can hang at the "Deferred Images ... (I let it run overnight lke that - nothing ever downloaded), or, in a few attempts, it got as far as "Step 2 / 3 ..." but mostly never finishing.  A few times, I've gotten downloads that never finish downloading, like this one: 

![image](https://user-images.githubusercontent.com/74366614/98959932-cbbf2a80-24d1-11eb-80ee-ea00b7860b61.png)

Twitter has been throwing obstacles at the ability to save timelines for a couple years now (my guess, unintentionally, but who knows) and it would seem to have thrown one SingleFile can't get past.  I've spent hours trying different things with no luck at all.  My “select all” and scrolling up or down before engaging SingleFile fallback option hasn’t helped at all.   I’m hoping you might have some suggestions I haven't tried.  

- Jim

**Comment by gildas-lormeau** [comment]
Hi Jim,

Actually, I noticed the issue but I was not able to identify the reason of the problem. It should be fixed in the version 1.18.35 I just uploaded. It should be available for download in the coming 24 hours. The update for SingleFileZ should be already available for download (in Firefox). Make sure the option "Images > zoom out the page" is enabled.

**Comment by gildas-lormeau** [comment]
My bad, it doesn't seem to be really fixed... I should be able to find a solution though.

**Comment by gildas-lormeau** [comment]
Meanwhile, you can circumvent this issue by disabling "Images > save deferred images" in SingleFile and running the following code in the console (F12) before saving the page :
```js
const height = parseInt(getComputedStyle(document.querySelector("main")).height);
document.documentElement.__defineGetter__("clientHeight", () => height);
```
I think this could be automated with a userscript.

**Comment by JimOutsideDC** [comment]
Thank you, Gildas.   Unfortunately, I’m afraid my coding skills are badly … outdated.  I have no doubt your suggestion would work — that is, if I understood it and knew how to implement it.  But, since I don’t, out of concern for my Mac’s safety and wellbeing (cause I’m sure I’d blow it up), I’ll just sit here waiting patiently until you say everything is fixed.  ;)

**Comment by gildas-lormeau** [comment]
I fixed the issue in the version 1.18.36+ which should be available for download in Firefox in ~ 24/48 hours. You won't need to enable the option `Images > zoom out the page` anymore. It should now work with the default settings of SingleFile. The fix is also available in the version 0.6.26 of SingleFileZ which should be already available for download in Firefox.
